src/client/opamTreeCommand.ml

```diff
@@ -378,10 +378,10 @@ let print_solution st new_st missing solution =
 
 (** Setting states for building *)
 
-let get_universe tog st =
+let get_universe ~all tog st =
   let OpamListCommand.{doc; test; dev_setup; dev; _} = tog in
   OpamSwitchState.universe st ~doc ~test ~dev_setup ~force_dev_deps:dev
-    ~requested:st.installed
+    ~requested:(if all then st.packages else st.installed)
```

Comment:
tiny optimisation, i don't know if it worth keeping it.

Reply:
This conflicts with https://github.com/ocaml/opam/pull/5919

Reply:
PRs need to be queued or merged together (their content, not on master :)). this one resolve only the `--with-test --no-switch` issue while it keeps the faulty behaviour of retrieving with-test/doc dependencies for all packages.